ballerina/auth_desugar.bal

```diff
@@ -54,89 +54,122 @@ public isolated function authenticateResource(Service serviceRef, string methodN
-                }
+        if scheme is AUTH_SCHEME_BASIC {
+            if config is FileUserStoreConfigWithScopes {
+                authResult = authenticateWithFileUserStore(config, header);
+            } else if config is LdapUserStoreConfigWithScopes {
+                authResult = authenticateWithLdapUserStoreConfig(config, header);
             }
-            jwt:Payload|Unauthorized authn = handler.authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is jwt:Payload {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler.authorize(authn, scopes);
-                    return authz;
-                }
-                return;
+        } else if scheme is AUTH_SCHEME_BEARER {
+            if config is JwtValidatorConfigWithScopes {
+                authResult = authenticateWithJwtValidatorConfig(config, header);
+            } else if config is OAuth2IntrospectionConfigWithScopes {
+                authResult = authenticateWithOAuth2IntrospectionConfig(config, header);
             }
+        }
+        if authResult is () || authResult is Forbidden {
+            return authResult;
+        }
+    }
+    return authResult;
+}
+
+isolated function authenticateWithFileUserStore(FileUserStoreConfigWithScopes config, string header)
+                                                returns Unauthorized|Forbidden? {
+    ListenerFileUserStoreBasicAuthHandler handler;
+    lock {
+        string key = config.fileUserStoreConfig.toString();
+        if authHandlers.hasKey(key) {
+            handler = <ListenerFileUserStoreBasicAuthHandler> authHandlers.get(key);
         } else {
-            // Here, config is OAuth2IntrospectionConfigWithScopes
-            ListenerOAuth2Handler handler;
-            lock {
-                string key = config.oauth2IntrospectionConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerOAuth2Handler> authHandlers.get(key);
-                } else {
-                    handler = new(config.oauth2IntrospectionConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
-            }
-            oauth2:IntrospectionResponse|Unauthorized|Forbidden auth = handler->authorize(header, config?.scopes);
-            if auth is oauth2:IntrospectionResponse {
-                return;
-            } else if auth is Forbidden {
-                return auth;
-            }
+            handler = new(config.fileUserStoreConfig.cloneReadOnly());
+            authHandlers[key] = handler;
+        }
+    }
+    auth:UserDetails|Unauthorized authn = handler.authenticate(header);
+    string|string[]? scopes = config?.scopes;
+    if authn is auth:UserDetails {
+        if scopes is string|string[] {
+            Forbidden? authz = handler.authorize(authn, scopes);
+            return authz;
+        }
+        return;
+    } else {
```

Comment:
I think now we can simply return as the type narrowing removes the `auth:UserDetails` type at this stage. Cant we?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-http/pull/794/commits/5ab10c20d35dd80a266c22fb5fc064d92f14f7d6